File: tests/py_ymcell/read_mislib_test.py
#! /usr/bin/env python3

"""py_ym_cell 用のテストプログラム

:file: read_mislib_test.py
:author: Yusuke Matsunaga (松永 裕介)
:copyright: Copyright (C) 2022 Yusuke Matsunaga, All rights reserved.
"""

import sys
import io
import ym_cell

assert len(sys.argv) == 3
filename = sys.argv[1]
exp_filename = sys.argv[2]
lib = ym_cell.ClibCellLibrary.read_mislib(filename)
lines = lib.to_string_list()

with open(exp_filename, "rt") as exp_f:
    exp_lines = exp_f.readlines()

    assert len(lines) == len(exp_lines)
    n = len(lines)
    for i in range(n):
        line = lines[i]
        exp_line = exp_lines[i].rstrip('\n')
        if line != exp_line:
            print(f'"lines[{i}]     = {lines[i]}"')
            print(f'"exp_lines[{i}] = {exp_lines[i]}"')
            assert False

File: tests/programs/read_mislib_test.cc

/// @file read_mislib_test.cc
/// @brief mislib 形式のファイル読み込みのテストプログラム
/// @author Yusuke Matsunaga (松永 裕介)
///
/// Copyright (C) 2022 Yusuke Matsunaga
/// All rights reserved.

#include "ym/ClibCellLibrary.h"

BEGIN_NAMESPACE_YM

int
read_mislib(
  int argc,
  char** argv
)
{
  string filename{TESTFILE};

  auto library = ClibCellLibrary::read_mislib(filename);

  library.display(cout);

  return 0;
}

END_NAMESPACE_YM


int
main(
  int argc,
  char** argv
)
{
  return YM_NAMESPACE::read_mislib(argc, argv);
}
